.github/workflows/pr.yml

```diff
@@ -0,0 +1,41 @@
+---
+name: CI
+on:
+  push:
+    tags:
+      - v*
+    branches:
+      - master
+  pull_request:
+jobs:
+  golangci-lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: lint
+        uses: golangci/golangci-lint-action@v2.3.0
+        with:
+          version: v1.33.0
+
+  tests-on-unix:
+    needs: golangci-lint # run after golangci-lint action to not produce duplicated errors
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        golang:
+          - 1.14
+          - 1.15
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: ${{ matrix.golang }}
+      - uses: actions/cache@v2
```

Comment:
Can we please put names for each step? Currently, it looks confusing. Each step have `uses` but only one step have `name`. (except run tests)

Reply:
add to this but a name for `actions/checkout@v2` would be "checkout" which seems redundant

Reply:
For me checkout and cache (what?) were not very clear.

Step name suggestions:

- name `Checkout repository`, uses `actions/checkout@v2`
- name `Install Go`, uses `actions/setup-go@v2`
- name `Cache dependencies and build outputs`, uses `actions/cache@v2`